Code/autopkglib/URLTextSearcher.py

```diff
@@ -81,24 +81,32 @@ class URLTextSearcher(URLGetter):
 
     description = __doc__
 
+    match_message = "Found matching text"
+    no_match_message = "No match found on URL"
+
     def prepare_curl_cmd(self):
         """Assemble curl command and return it."""
         curl_cmd = super().prepare_curl_cmd()
         self.add_curl_common_opts(curl_cmd)
         curl_cmd.append(self.env["url"])
         return curl_cmd
 
-    def re_search(self, content):
-        """Search for re_pattern in content"""
+    def prepare_re_flags(self):
+        """Create flag varible for re.compile"""
         flag_accumulator = 0
         for flag in self.env.get("re_flags", {}):
             if flag in re.__dict__:
                 flag_accumulator += re.__dict__[flag]
-        re_pattern = re.compile(self.env["re_pattern"], flags=flag_accumulator)
+        return flag_accumulator
+
+    def re_search(self, content):
+        """Search for re_pattern in content"""
+
+        re_pattern = re.compile(self.env["re_pattern"], flags=self.prepare_re_flags())
         match = re_pattern.search(content)
 
         if not match:
-            raise ProcessorError(f"No match found on URL: {self.env['url']}")
+            raise ProcessorError(f"{self.no_match_message}: {self.env['url']}")
```

Comment:
Not sure I really get the point of this, but okay.

Reply:
`self.no_match_message` variable is used in the test class. When there is no match `ProcessorError` gets raised. Since it's quite generic exception I check for `No match found on URL`. If someone decide to reword this string test breaks. So I was thinking to put into variable I can reuse.

I am not sure if this is a good idea or anti pattern. If you think it's bad approach I can drop the 1331358 commit and update the test.

Reply:
> If someone decide to reword this string test breaks. So I was thinking to put into variable I can reuse.

That's a fair argument, but if that's the case, we should codify it even more:
* URLTextSearcher should put that output text into a constant
* URLTextSearcher should raise an exception with the contents of the constant when no match is found
* This unit test should import that constant, and use that for the test